lib/request-fragment.js

```diff
@@ -57,7 +57,12 @@ module.exports = filterHeaders => (
             fragmentRequest.setTimeout(timeout, fragmentRequest.abort);
         }
         fragmentRequest.on('response', response => {
-            if (response.statusCode >= 500) {
+            const isError500 = response.statusCode >= 500;
+            const isNonPrimaryAndNon200 =
+                (response.statusCode < 200 || response.statusCode >= 300) &&
+                !fragmentAttributes.primary;
+
+            if (isError500 || isNonPrimaryAndNon200) {
                 reject(new Error('Internal Server Error'));
```

Comment:
Could you PLEASE do a better job at error reporting? With this code we won't be able to see which kind of error we really had... Would be great if you could report at least response status code & body (if available).

Reply:
I know and I did it previously(https://github.com/StyleT/tailorx/blob/a90ad6ea82b72795d4ce5bcaa3dd95689ba91cfd/lib/request-fragment.js#L63) but decided to revert...

ok, I will do.

Reply:
Also keep in mind that `Internal Server Error!` text in message is relevant only for 5XX errors.